### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by garlicnation**
Clean up some logic around loading resources. CSS/js resources that fail to load are now automatically excluded.

### CODE REVIEW

File: src/bundler.ts [hunk 1]
```diff
@@ -483,25 +510,7 @@ class Bundler {
     // feature as a value indicating the inlining of the Import has not yet
     // occurred or a value of null indicating that <link> tags referencing it
     // should be removed from the document.
```
Comment (usergenic) [inline src/bundler.ts]: This comment should be changed or removed now that variable changed.

File: src/bundler.ts [hunk 2]
```diff
@@ -293,24 +321,23 @@ class Bundler {
...
     }
 
-    // If we've just inlined it or otherwise have seen it before, we can remove
-    // the <link> tag.
-    if (importMap.get(resolvedUrl) === null) {
+    if (reachedImports.has(resolvedUrl)) {
       dom5.remove(htmlImport);
     }
 
     // If we've never seen this import before, lets put it on the map as null so
```
Comment (usergenic) [inline src/bundler.ts]: Change to
> If we've never seen this import before, lets add it to the set so we will deduplicate if we encounter it again.

File: src/bundler.ts [hunk 3]
```diff
@@ -293,24 +321,23 @@ class Bundler {
...
 
       astUtils.insertAllBefore(importParent, htmlImport, importDoc.childNodes!);
     }
 
-    // If we've just inlined it or otherwise have seen it before, we can remove
-    // the <link> tag.
-    if (importMap.get(resolvedUrl) === null) {
+    if (reachedImports.has(resolvedUrl)) {
       dom5.remove(htmlImport);
     }
```
Comment (usergenic) [inline src/bundler.ts]: Should the following `if (!reachedImports.has(resolvedUrl)` maybe be converted to an else clause of this if statement?

File: src/bundler.ts [hunk 4]
```diff
@@ -542,10 +551,8 @@ class Bundler {
             newUrl, 'html-import', importDoc, undefined, undefined, newNode, [
             ]);
```
Comment (usergenic) [inline src/bundler.ts]: I don't think this `const im = new Import` thing needs to exist anymore...?

File: src/bundler.ts [hunk 5]
```diff
@@ -568,32 +575,35 @@ class Bundler {
       }
     });
 
+    const reachedImports = new Set<UrlString>([url]);
+
     // Inline all HTML Imports.  (The inlineHtmlImport method will discern how
     // to handle them based on the state of the importMap.)
```
Comment (usergenic) [inline src/bundler.ts]: Fix comment; importMap no longer a thing.

**Comment by usergenic** [comment]
I meant to submit as "Request Changes" but I don't see an undo for that on Github.  Most of the issues are minor so keeping this as LGTM is fine-- just have a look at the comments I made.

File: src/bundler.ts (hunk 3 above)
Comment (garlicnation) [inline src/bundler.ts]: Done.

File: src/bundler.ts (hunk 2 above)
Comment (garlicnation) [inline src/bundler.ts]: Done.

File: src/bundler.ts (hunk 1 above)
Comment (garlicnation) [inline src/bundler.ts]: Done.

File: src/bundler.ts (hunk 4 above)
Comment (garlicnation) [inline src/bundler.ts]: Done

File: src/bundler.ts (hunk 5 above)
Comment (garlicnation) [inline src/bundler.ts]: Done.

File: src/bundler.ts [hunk 6]
```diff
@@ -243,33 +254,50 @@ class Bundler {
...
-    //   dom5.setAttribute(htmlImport, 'href', bundleUrl);
-    // }
+    if (!bundleUrl) {
+      if (reachedImports.has(resolvedUrl)) {
+        dom5.remove(htmlImport);
+        return;
+      } else {
+        reachedImports.add(resolvedUrl);
+      }
+      // dom5.remove(htmlImport);
```
Comment (usergenic) [inline src/bundler.ts]: remove commented-out code.

Comment (garlicnation) [inline src/bundler.ts]: Done

**Pull request merged**